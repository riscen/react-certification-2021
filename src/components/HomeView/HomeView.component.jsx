import React from 'react';
import { HomeViewContainer } from './HomeView.modules';

import VideoCard from '../VideoCard';

function HomeView(props) {
  const { youtubeData } = props;
  const videos =
    youtubeData !== undefined && 'items' in youtubeData ? youtubeData.items : [];

  return (
    <HomeViewContainer className="home-view-container">
      {videos
        .filter((videoInfo) => videoInfo.id.kind.includes('video'))
        .map((videoInfo) => {
          return (
            <VideoCard
              title={videoInfo.snippet.title}
              thumbnail={videoInfo.snippet.thumbnails.medium.url}
              description={videoInfo.snippet.description}
              videoId={videoInfo.id.videoId}
              key={videoInfo.id.videoId}
            />
          );
        })}
    </HomeViewContainer>
  );
}

export default HomeView;
