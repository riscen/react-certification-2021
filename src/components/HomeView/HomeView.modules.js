import styled from 'styled-components';

export const HomeViewContainer = styled.div`
  background-color: white;
  display: flex;
  flex-wrap: wrap;
  font-size: 18px;
  height: calc(100vh - 3.5em);
  justify-content: center;
  overflow-x: hidden;
`;
