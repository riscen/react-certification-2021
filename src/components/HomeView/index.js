export { default } from './HomeView.component';
export { HomeViewContainer } from './HomeView.modules';
