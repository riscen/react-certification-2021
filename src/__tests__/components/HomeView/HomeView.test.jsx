import React from 'react';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import HomeView, { HomeViewContainer } from '../../../components/HomeView';

describe('HomeView', () => {
  beforeEach(() => {
    configure({ adapter: new Adapter() });
  });

  it('Renders HomeView', () => {
    const wrapper = mount(<HomeView youtubeData={{}} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  it('Renders HomeViewContainer', () => {
    const wrapper = mount(<HomeView youtubeData={{}} />);
    expect(wrapper.find(HomeViewContainer)).toBeDefined();
    wrapper.unmount();
  });
});
