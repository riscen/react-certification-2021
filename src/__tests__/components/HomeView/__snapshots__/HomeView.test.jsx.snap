// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView Renders HomeView 1`] = `ReactWrapper {}`;
